files/get-random-osd.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python
+
+from __future__ import print_function
+
+import subprocess
+import json
+import argparse
+import random
+import socket
+
+_is_rack = False
+
+def get_map():
+    cmd = ['ceph', 'osd', 'crush', 'tree', '--format', 'json']
+    ceph_osd_crush_tree_json = subprocess.check_output(cmd)
+    ceph_osd_crush_tree = json.loads(ceph_osd_crush_tree_json)
+
+    osd_map = {}
+    global _is_rack
+
+    for top_item in ceph_osd_crush_tree:
+        if top_item['type'] == 'root':
+            for l2_item in top_item['items']:
+                if l2_item['type'] == 'rack':
+                    _is_rack = True
+                    osd_map[l2_item['name']] = {}
+                    for l3_item in l2_item['items']:
+                        if l3_item['type'] == 'host':
+                            osd_map[l2_item['name']][l3_item['name']] = []
+                            for osd in l3_item['items']:
+                                if osd['type'] == 'osd':
+                                    osd_map[l2_item['name']][l3_item['name']].append(osd['id'])
+                elif l2_item['type'] == 'host':
+                    _is_rack = False
+                    osd_map[l2_item['name']] = []
+                    for l3_item in l2_item['items']:
+                        if l3_item['type'] == 'osd':
+                            osd_map[l2_item['name']].append(l3_item['id'])
+    return osd_map
+
+def get_random_osd(num, same_bucket=False):
+
+    osd_map = get_map()
+    random_bucket = []
+    osd = []
+    random_osd = []
+
+    for bucket in osd_map:
+        random_bucket.append(bucket)
+    random.shuffle(random_bucket)
+
+    if same_bucket:
+        bucket = random_bucket[0]
+        
+        if _is_rack:
+            for host in osd_map[bucket]:
+                osd.extend(osd_map[bucket][host])
+        else:
+            osd = osd_map[bucket]
+
+    else:
+        for bucket in random_bucket:
+          if _is_rack:
+              for host in osd_map[bucket]:
+                  osd.extend(osd_map[bucket][host])
+          else:
+              osd.extend(osd_map[bucket])
+
+    random.shuffle(osd)
+    random_osd = osd[:num]
+
+    return random_osd
+
+def osd_to_host_ip(osd_id):
```

Comment:
we can retrieve those information from 'ceph osd dump', I think it is more reliable.
otherwise, we should config the /etc/hosts properly?　

or what we need here is to obtain the management network, not storage network?  if so, that is ok.